### PR TITLE
spring: refactor

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/spring/default.nix [hunk 1]
```diff
@@ -40,14 +40,13 @@ stdenv.mkDerivation rec {
...
-    xorg.libX11 xorg.libXcursor libGLU libGL glew asciidoc libxslt docbook_xsl curl makeWrapper
-    docbook_xsl_ns systemd libunwind which minizip ]
+  nativeBuildInputs = [ cmake makeWrapper docbook_xsl docbook_xsl_ns asciidoc ];
+  buildInputs = [ lzma boost libdevil zlib p7zip openal libvorbis freetype SDL2
+    xorg.libX11 xorg.libXcursor libGLU libGL glew curl
+    systemd libunwind which minizip ]
     ++ stdenv.lib.optional withAI jdk
     ++ stdenv.lib.optional withAI python;
 
-  enableParallelBuilding = true;
```
Comment (domenkozar) [inline pkgs/games/spring/default.nix:49]: Why?

**Review by domenkozar — CHANGES_REQUESTED** [review]

File: pkgs/games/spring/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/games/spring/default.nix:49]: This is set by cmake.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108062` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>spring</li>
<li>springLobby</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

springLobby.log:
<details>
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/games/spring/springlobby.nix:41:5:

```
   |
41 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
[1;33mwarning: unnecessary-parallel-building[0m
Meson, CMake and qmake already set `enableParallelBuilding = true` by default so it is not necessary.

Near pkgs/games/spring/springlobby.nix:29:3:

```
   |
29 |   enableParallelBuilding = true;
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unnecessary-parallel-building.md
</details>
spring.log:
<details>
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/games/spring/default.nix:60:5:

```
   |
60 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by siraben** [comment]
Maybe merge now, when the package author responds I'll do another PR updating license.

**Pull request merged**